### PR TITLE
Snip workflow logs 

### PR DESCRIPTION
**Pull request by Yancey0623**
related issue #1819 

TODO: update the design doc to snip HTML context and query result from Pod logs.

### CODE REVIEW

File: pkg/argo/fetch.go [hunk 1]
```diff
@@ -126,6 +126,19 @@ func Fetch(req *pb.FetchRequest) (*pb.FetchResponse, error) {
...
+
+		// snip the pod logs when it complete
+		// TODO(yancey1989): fetch the pod logs using an iteration way
+		// to avoid fetching a large number of logs in once.
+		podLogs, e := k8sReadPodLogs(pod.ObjectMeta.Name, "main", "", false)
+		if e != nil {
+			return nil, e
+		}
+		snipLogs, e := snipPodLogs(podLogs)
+		if e != nil && len(snipLogs) >= 0 {
```
Comment (weiguoz) [inline pkg/argo/fetch.go]: `e == nil` ?

File: pkg/argo/fetch.go [hunk 2]
```diff
@@ -144,6 +158,21 @@ func Fetch(req *pb.FetchRequest) (*pb.FetchResponse, error) {
 	return newFetchResponse(newFetchRequest(req.Job.Id, stepGroupName, newStepPhase), eof, logs), nil
 }
 
+func isHTMLCode(log string) bool {
+	return strings.HasPrefix(log, "<div")
```
Comment (typhoonzero) [inline pkg/argo/fetch.go]: Maybe use regex in the future to match "<div .....> .... </div>"

File: pkg/argo/fetch.go [hunk 3]
```diff
@@ -144,6 +158,21 @@ func Fetch(req *pb.FetchRequest) (*pb.FetchResponse, error) {
...
+
+func snipPodLogs(podLogs []string) ([]string, error) {
+	snipLogs := []string{}
+	for _, log := range podLogs {
+		if isHTMLCode(log) {
+			snipLogs = append(snipLogs, log)
+		}
+		// TODO(yancey1989): snip query result from logs
+	}
+	return snipLogs, nil
```
Comment (weiguoz) [inline pkg/argo/fetch.go:179]: Whether it will appear in the pod logs like:
```html
<div>
   2nd line
</div>
```

Comment (Yancey0623) [reply pkg/argo/fetch.go:179]: May not, here follows:
https://github.com/sql-machine-learning/sqlflow/blob/eed73c55f9531279263dca36a5139275b6abc5ce/pkg/sql/submitter.go#L214

but I think more complex HTML code would be used in the future.

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**